File: lec_map.py
import sys

def main():
    motorways = {}

    motorways["M1"] = (193.5, 1959)
    motorways["M2"] = (25.7, 1963)
    motorways["M3"] = (58.6, 1971)
    motorways["M4"] = (191.9, 1961)
    #                 [...]
    motorways["M898"] = (0.5, 1985)

    print( 'The %s is %0.1f miles long' % ("M4", motorways["M4"][0]) )
    print( 'The %s opened in %d' % ("M898", motorways["M898"][1]) )

if __name__ == '__main__':
    sys.exit(main())

File: lec_map.cpp
#include <iostream>
#include <map>
using namespace std;

int main()
{
	map< string, pair<float,int> > motorways;

	motorways.emplace( "M1", make_pair<float,int>(193.5, 1959) );
	//                 [...]
	motorways.emplace( "M898", make_pair<float,int>( 0.5, 1985 ) );

	cout << "The " << "M1" << " is " << 
		motorways.find("M1")->second.first << " miles long" << endl;

	cout << "The " << "M898" << " opened in " << 
		motorways.find("M898")->second.second << endl;

	return 0;
}